src/key.js

```diff
@@ -261,30 +261,37 @@ Key.prototype.armor = function() {
 };
 
 /**
- * Returns the signature that has the latest creation date, while ignoring signatures created in the future.
+ * Returns the valid and non-expired signature that has the latest creation date, while ignoring signatures created in the future.
  * @param  {Array<module:packet.Signature>} signatures  List of signatures
  * @param  {Date}                           date        Use the given date instead of the current time
- * @returns {module:packet.Signature} The latest signature
+ * @returns {Promise<module:packet.Signature>} The latest valid signature
+ * @async
  */
-function getLatestSignature(signatures, date=new Date()) {
-  let signature = signatures[0];
-  for (let i = 1; i < signatures.length; i++) {
-    if (signatures[i].created >= signature.created &&
-       (signatures[i].created <= date || date === null)) {
+async function getLatestValidSignature(signatures, primaryKey, dataToVerify, date=new Date()) {
+  let signature;
+  for (let i = signatures.length - 1; i >= 0; i--) {
+    if (
+      (!signature || signatures[i].created >= signature.created) &&
+      // check binding signature is not expired (ie, check for V4 expiration time)
+      !signatures[i].isExpired(date) &&
+      // check binding signature is verified
+      (signatures[i].verified || await signatures[i].verify(primaryKey, dataToVerify))
+    ) {
       signature = signatures[i];
     }
   }
   return signature;
 }
 
-function isValidSigningKeyPacket(keyPacket, signature, date=new Date()) {
+function isValidSigningKeyPacket(keyPacket, signature) {
+  if (!signature.verified || signature.revoked !== false) { // Sanity check
+    throw new Error("Signature wasn't verified before calling isValidSigningKeyPacket");
```

Comment:
Instead of doing this maybe we can just make this a sub-function of `Key.prototype.getSigningKey` as it isn't used anywhere else and doesn't seem generically useful